### PR TITLE
Define sentinel file to opt out of native tool installation

### PR DESCRIPTION
**Pull request by hoyosjs**
In CoreCLR we are manually change scripts under `eng/common` on every update to remove native tool installation as it indefinitely hangs our CI for Linux and OSX legs. Then env variable work around exists, but we use the script in many different places and it is currently affecting dev loops as well. This **workaround** allows to drop a file named  `disable-nativetool-install.workaround` under the `eng` folder to skip the automatic acquisition of native tools until #2673 gets resolved.

Are we ok with these names and logic?

@RussKeldorph @jkotas

### CODE REVIEW

File: eng/common/build.ps1 [hunk 1]
```diff
@@ -126,7 +126,8 @@ try {
     . $configureToolsetScript
   }
 
-  if (($restore) -and ($null -eq $env:DisableNativeToolsetInstalls)) {
+  $nativeToolsOptOutFile = Join-Path $EngRoot "disable-nativetool-install"
```
Comment (chcosta) [inline eng/common/build.ps1]: We should add a note that this is a temporary fix (ie, teams shouldn't take a dependency on it), until native toolset has been redesigned, and link to the relevant issue.

File: eng/common/build.sh [hunk 2]
```diff
@@ -209,7 +209,8 @@ if [[ -n "${useInstalledDotNetCli:-}" ]]; then
   use_installed_dotnet_cli="$useInstalledDotNetCli"
 fi
 
-if [[ "$restore" == true && -z ${DisableNativeToolsetInstalls:-} ]]; then
+native_tools_opt_out_file="$eng_root/disable-nativetool-install.workaround"
```
Comment (chcosta) [inline eng/common/build.sh]: same, make a note.

Why is one named `disable-nativetool-install` and the other named `disable-nativetool-install.workaround`?  Shouldn't they be the same.  But actually, do you want a Windows version and a Unix version.  ie `disable-nativetool-install-unix` and `disable-nativetool-install-windows`?

Comment (chcosta) [reply eng/common/build.sh]: Given the pre-existing variable is named `DisableNativeToolsetInstalls`, maybe name the files `disable-native-toolset-installs-blah` just so it's somewhat consistent

File: eng/common/build.ps1 [hunk 3]
```diff
@@ -126,7 +126,8 @@ try {
     . $configureToolsetScript
   }
 
-  if (($restore) -and ($null -eq $env:DisableNativeToolsetInstalls)) {
+  $nativeToolsOptOutFile = Join-Path $EngRoot "disable-nativetool-install.workaround"
+  if (($restore) -and ($null -eq $env:DisableNativeToolsetInstalls) -and (-not Test-Path $nativeToolsOptOutFile)) {
```
Comment (chcosta) [inline eng/common/build.ps1]: I think this needs to be `(-not (Test-Path $nativeToolsOptOutFile))`, but a local repro should show the same failure as you're hitting in CI..

File: eng/common/build.ps1 [hunk 4]
```diff
@@ -126,7 +126,8 @@ try {
     . $configureToolsetScript
   }
 
-  if (($restore) -and ($null -eq $env:DisableNativeToolsetInstalls)) {
+  $nativeToolsOptOutFile = Join-Path $EngRoot "disable-nativetool-install.workaround"
```
Comment (chcosta) [inline eng/common/build.ps1]: probably name this variable `nativeToolsetOptOutFile` for consistency.

File: eng/common/build.sh (hunk 2 above)
Comment (hoyosjs) [reply eng/common/build.sh]: Looks like I didn't push the last changes, sorry. Also, indeed I made it one file. We wanted to be explicit about using this feature, and in CI only. If there is some misalignment here with Arcade's principles I can change this easily. I'm not married to anything in this PR, it is a throw-away workaround.

Comment (chcosta) [reply eng/common/build.sh]: I thought you wanted to keep Windows, but disable Unix.

Regardless, if you want this to be enabled in CI only, there's a `$ci` variable that can be used to reference context.

Comment (hoyosjs) [reply eng/common/build.sh]: We explicitly call them on Windows
https://github.com/dotnet/coreclr/blob/404e82becbadbd3def532047e5504e1313c1766a/eng/build-job.yml#L101
In Linux, Docker solves our issues, and in Mac, we have another solution. We don't really want dev scenarios doing any of this.

**Comment by hoyosjs** [comment]
If we are ok with this I'll add the logic and an issue on CoreCLR as well as add a note in the issue so that we clean this up for sure as well.

**Comment by hoyosjs** [comment]
@chcosta, if you have no more feedback, can we merge this please?

File: eng/common/build.ps1 [hunk 5]
```diff
@@ -126,7 +126,12 @@ try {
     . $configureToolsetScript
   }
 
-  if (($restore) -and ($null -eq $env:DisableNativeToolsetInstalls)) {
+  # This is a sentinel file to work around hangs caused by native tool installation.
+  # This is currently tracked by https://github.com/dotnet/arcade/issues/2673
+  # Don't take a dependency on this as this is expected to be temporary unless the work
+  # to revert this is being actively tracked along the mentioned issue.
```
Comment (chcosta) [inline eng/common/build.ps1]: nit: This sentence is awkward.

> Don't take a dependency on this as this is expected to be temporary unless the work to revert this is being actively tracked along the mentioned issue.

File: eng/common/build.sh [hunk 6]
```diff
@@ -209,7 +209,12 @@ if [[ -n "${useInstalledDotNetCli:-}" ]]; then
   use_installed_dotnet_cli="$useInstalledDotNetCli"
 fi
 
-if [[ "$restore" == true && -z ${DisableNativeToolsetInstalls:-} ]]; then
+# This is a sentinel file to work around hangs caused by native tool installation.
+# This is currently tracked by https://github.com/dotnet/arcade/issues/2673
+# Don't take a dependency on this as this is expected to be temporary unless the work
```
Comment (chcosta) [inline eng/common/build.sh]: nit: awkward sentence

**Comment by chcosta** [comment]
@tmat, thoughts?

**Comment by tmat** [comment]
Why not set a variable in `eng/configure-toolset.ps1`?

**Comment by hoyosjs** [comment]
> Why not set a variable in `eng/configure-toolset.ps1`?

That would work for us. Does this get called in all cases?

**Comment by tmat** [comment]
@hoyosjs See https://github.com/dotnet/arcade/blob/master/eng/common/build.sh#L199-L203

**Comment by hoyosjs** [comment]
Thanks.

**Pull request closed without merging**